kombu/transport/redis_cluster.py

```diff
@@ -0,0 +1,1101 @@
+
+    def get_table(self, exchange):
+        key = self.keyprefix_queue % exchange
+        with self.conn_or_acquire() as client:
+            values = client.smembers(key)
+            if not values:
+                raise InconsistencyError(NO_ROUTE_ERROR.format(exchange, key))
+            return [tuple(bytes_to_str(val).split(self.sep)) for val in values]
+
+    def _purge(self, queue):
+        with self.conn_or_acquire() as client:
+            with client.pipeline(False) as pipe:
+                for pri in PRIORITY_STEPS:
+                    priq = self._q_for_pri(queue, pri)
+                    pipe = pipe.llen(priq).delete(priq)
+                sizes = pipe.execute()
+                return sum(sizes[::2])
+
+    def close(self):
+        self._closing = True
+        self._disconnect_pools()
+        if not self.closed:
+            # remove from channel poller.
+            self.connection.cycle.discard(self)
+
+            # delete fanout bindings
+            for queue in self._fanout_queues:
+                if queue in self.auto_delete_queues:
+                    self.queue_delete(queue)
+
+            self._close_clients()
+
+        super(Channel, self).close()
+
+    def _close_clients(self):
+        # Close connections
+        for attr in 'client', 'subclient':
+            try:
+                self.__dict__[attr].connection.disconnect()
+            except (KeyError, AttributeError, self.ResponseError):
+                pass
+
+    def _prepare_virtual_host(self, vhost):
+        if not isinstance(vhost, numbers.Integral):
+            if not vhost or vhost == '/':
+                vhost = DEFAULT_DB
+            elif vhost.startswith('/'):
+                vhost = vhost[1:]
+            try:
+                vhost = int(vhost)
+            except ValueError:
+                raise ValueError(
+                    'Database is int between 0 and limit - 1, not {0}'.format(
+                        vhost,
+                    ))
+        return vhost
+
+    def _filter_tcp_connparams(self, socket_keepalive=None,
+                               socket_keepalive_options=None, **params):
+        return params
+
+    def _parse_hosts(self, hostname):
+
+        """
+        hostname:
+        redis-cluster://username:password@127.0.0.1:30001/0?alts=127.0.0.1:30002'
+        """
+        _, host, port, _, password, path, query = _parse_url(hostname)
+
+        hosts = [
+            {
+                'host': host or '127.0.0.1',
+                'port': port or DEFAULT_PORT,
+            }
+        ]
+
+        if query.get('alts'):
+            alts = query['alts'].split(',')
+
+            for url in alts:
+                _, host, port, _, _, _, _ = _parse_url('://' + url)
+
+                hosts.append({
+                    'host': host or '127.0.0.1',
+                    'port': port or DEFAULT_PORT,
+                })
+
+        return hosts, password, path
+
+    def _connparams(self, _r210_options=(
```

Comment:
This tuple should probably be moved to a constant?

Reply:
will check